src/PrestaShopBundle/Entity/ApiAccess.php

```diff
@@ -30,60 +30,68 @@
 
 use Doctrine\ORM\Mapping as ORM;
 use PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Model\ApiAccessInterface;
-use PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Model\AuthorizedApplicationInterface;
 use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+use Symfony\Component\Validator\Constraints as Assert;
 
 /**
  * @ORM\Entity(repositoryClass="PrestaShopBundle\Entity\Repository\ApiAccessRepository")
  * @ORM\Table()
- * @UniqueEntity("name")
  *
  * @experimental
  */
+#[UniqueEntity('clientId')]
+#[UniqueEntity('clientName')]
 class ApiAccess implements ApiAccessInterface
 {
     /**
-     * @var int
-     *
      * @ORM\Id
-     * @ORM\Column(name="id_api_access", type="integer", options={"unsigned":true})
+     * @ORM\Column(name="id_api_access", type="integer", options={"unsigned": true})
      * @ORM\GeneratedValue(strategy="AUTO")
      */
-    private $id;
+    #[Assert\Type('integer')]
+    #[Assert\Positive]
+    private int $id;
 
     /**
-     * @var string
-     *
      * @ORM\Column(name="client_id", type="string", length=255)
      */
-    private $clientId;
+    #[Assert\NotBlank]
+    #[Assert\Type('string')]
+    private string $clientId;
 
     /**
-     * @var string
-     *
-     * @ORM\Column(name="client_secret", type="string", length=255)
+     * @ORM\Column(name="client_name", type="string")
      */
-    private $clientSecret;
+    #[Assert\NotBlank]
+    #[Assert\Type('string')]
+    private string $clientName;
 
     /**
-     * @var AuthorizedApplicationInterface
+     * We make the secret nullable for the moment because it prevents the first step of the feature to be implemented.
      *
-     * @ORM\ManyToOne(targetEntity=AuthorizedApplication::class)
-     * @ORM\JoinColumn(name="id_authorized_application", referencedColumnName="id_authorized_application", nullable=false, onDelete="CASCADE")
+     * @ORM\Column(name="client_secret", type="string", length=255, nullable=true)
      */
-    private $authorizedApplication;
+    #[Assert\Type(['string', 'null'])]
+    private string $clientSecret;
 
     /**
-     * @var bool
-     *
-     * @ORM\Column(name="active", type="boolean")
+     * @ORM\Column(name="enabled", type="boolean")
      */
-    private $active;
+    #[Assert\NotBlank]
+    #[Assert\Type('bool')]
+    private bool $enabled;
 
     /**
      * @ORM\Column(name="scopes", type="array")
      */
-    private $scopes = [];
+    #[Assert\Type('array')]
+    private array $scopes = [];
+
+    /**
+     * @ORM\Column(name="description", type="string")
+     */
+    #[Assert\Type('string')]
+    private string $description = '';
```

Comment:
I don't know what type Doctrine uses by default for long text like this and if there's a limit in the end In case you need one for product description we use `public const MAX_DESCRIPTION_LENGTH = 21844;` so it should be more than enough here as well